lzma/Wrapper-CPP/C7Zip.cpp

```diff
@@ -209,22 +209,26 @@ bool C7Zip::AddPath(const std::wstring& path)
 
     // set the compression properties
     bool                         encryptHeaders = (m_compressionFormat == CompressionFormat::SevenZip && !m_password.empty());
-    const size_t                 numProps       = 2;
+    const size_t                 numProps       = 4;
     const wchar_t*               names[numProps];  // = { L"x" };
     NWindows::NCOM::CPropVariant values[numProps]; // = { static_cast<UInt32>(m_compressionLevel) };
     names[0]  = L"x";
     values[0] = static_cast<UInt32>(m_compressionLevel);
+    names[1]  = L"tc"; // Stores Creation timestamps for files.
+    values[1] = true;
+    names[2]  = L"ta"; // Stores last Access timestamps for files.
+    values[2] = true;
     if (encryptHeaders)
```

Comment:
small optimization: instead of hard coded indexes, use an int and increase it for each name/value pair:

```
int propIndex = 0;
names[propIndex] = L"x";
values[propIndex++] = static_cast<UInt32>(m_compressionLevel);
....
```

also, is there a reason why you don't pass "tm" as well (to keep the last-modified timestamp)?

Reply:
I'll optimize as suggested. I did not put "tm" because the default is "true" for tm. That being said, i'll add "tm" so the wanted behavior remains in effect if the default ever changes.